hazelcast/src/main/java/com/hazelcast/map/impl/MapKeyLoader.java

```diff
@@ -317,7 +316,7 @@ private void sendKeysInBatches(MapStoreContext mapStoreContext, boolean replaceE
             // Re-ordering of in-flight operations can happen during a partition migration. We are waiting here
             // for all LoadAllOperation(s) to be ACKed by receivers and only then we send them the LoadStatusOperation
             // See https://github.com/hazelcast/hazelcast/issues/4024 for additional details
-            FutureUtil.waitWithDeadline(futures, KEY_DISTRIBUTION_TIMEOUT_MINUTES, TimeUnit.MINUTES);
+            FutureUtil.waitForever(futures);
```

Comment:
I also see counter examples in `FutureUtil`, some inconsistency exists between methods. 
To me your approach makes more sense. 👍